pgjdbc/src/main/java/org/postgresql/core/NativeQuery.java

```diff
@@ -27,13 +28,14 @@
   }
 
   public NativeQuery(String nativeSql) {
-    this(nativeSql, NO_BINDS);
+    this(nativeSql, NO_BINDS, false);
   }
 
-  public NativeQuery(String nativeSql, int[] bindPositions) {
+  public NativeQuery(String nativeSql, int[] bindPositions, boolean compatible) {
```

Comment:
On Mon, Apr 4, 2016 at 9:03 AM, Vladimir Sitnikov <notifications@github.com>
wrote:

> In pgjdbc/src/main/java/org/postgresql/core/NativeQuery.java
> <https://github.com/pgjdbc/pgjdbc/pull/491#discussion_r58400784>:
>
> >    }
> >
> > -  public NativeQuery(String nativeSql, int[] bindPositions) {
> > +  public NativeQuery(String nativeSql, int[] bindPositions, boolean compatible) {
>
> @polobo <https://github.com/polobo> , when you say "public API", you mean
> "public to pgjdbc, not public to everyone", don't you?
>

​I could see it being desirable for users to be able to access this
information but it wouldn't be a primary concern at this point.  I'm not
familiar enough with how one would even get access to a Native Query
instance to know how such truly public access would be supported.  The main
thing is that different parts of the pgjdbc implementation have access to
the results of the statement parsing.

David J.​